### DISCUSSION TITLE
How to run dev environment?

### DISCUSSION THREAD

**Discussion by iBobik** [Dave136/vue-email#137] [General]
Hi, I wanted to contribute, but end up with running playground locally, so I can test independently on my project.

Tried tu run it in the GitHub Codespace, so it will not be influenced by my local OS:

```
/workspaces/vue-email/playground (main) $ pnpm run dev

> playground@0.0.0 dev /workspaces/vue-email/playground
> vite


  VITE v5.0.10  ready in 321 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
10:14:58 PM [vite] Pre-transform error: Failed to resolve import "file:///workspaces/vue-email/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/lib/index.js" from "../dist/index.mjs". Does the file exist?
10:14:58 PM [vite] Pre-transform error: Failed to resolve import "file:///workspaces/vue-email/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/lib/index.js" from "../dist/index.mjs". Does the file exist? (x2)

warn - The `content` option in your Tailwind CSS configuration is missing or empty.
warn - Configure your content sources or your generated CSS will be missing styles.
warn - https://tailwindcss.com/docs/content-configuration
10:14:59 PM [vite] Pre-transform error: Failed to resolve import "file:///workspaces/vue-email/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/lib/index.js" from "../dist/index.mjs". Does the file exist? (x3)
10:14:59 PM [vite] Internal server error: Failed to resolve import "file:///workspaces/vue-email/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/lib/index.js" from "../dist/index.mjs". Does the file exist?
  Plugin: vite:import-analysis
  File: /workspaces/vue-email/dist/index.mjs:1:19
  1  |  import jiti from "file:///workspaces/vue-email/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/lib/index.js";
     |                    ^
  2  |  
  3  |  /** @type {import("/workspaces/vue-email/src/index")} */
      at formatError (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:63410:46)
      at TransformContext.error (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:63404:19)
      at normalizeUrl (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:61679:33)
      at async file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:61833:47
      at async Promise.all (index 0)
      at async TransformContext.transform (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:61754:13)
      at async Object.transform (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:63705:30)
      at async loadAndTransform (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:49386:29)
      at async viteTransformMiddleware (file:///workspaces/vue-email/node_modules/.pnpm/vite@5.0.10_@types+node@18.19.3/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:58987:32)
```

Could you please describe how exactly run everything I need for development, so I can transform some of my feature requests into pull requests? :-)

I am experienced with Vue and Vite, but I have no experience with pnpm, so I am lost in your workspaces config.

**Comment by Flowko** [comment]
hi, so to run the project, you have to build first, playground will only work after you build, so run build on the main directory, and when you go inside playground, running `pnpm dev` should work fine, let me know

**Comment by iBobik** [comment]
Hi, tried to build root first, but this was the result.Honza Pobořil💬Signalhonza.poboril.cz6. 1. 2024 v 14:20, Younes Barrad ***@***.***>:﻿
hi, so to run the project, you have to build first, playground will only work after you build, so run build on the main directory, and when you go inside playground, running pnpm dev should work fine, let me know

**Comment by Flowko** [comment]
hmm weird, technically all you have to do is `pnpm install`and `pnpm build`and go to playground, but ill have to recheck it on another folder to see, not sure if theres something wrong somewhere with pnpm

**Comment by iBobik** [comment]
Easiest is to start it in GH codespace, button in clone menu on the project page. You will test it in the clean docker container.Honza Pobořil💬Signalhonza.poboril.cz10. 1. 2024 v 15:26, Younes Barrad ***@***.***>:﻿
hmm weird, technically all you have to do is pnpm installand pnpm buildand go to playground, but ill have to recheck it on another folder to see, not sure if theres something wrong somewhere with pnpm

**Comment by Flowko** [comment]
yeah forgot about that, will do it when im free, will update here

**Comment by Flowko** [comment]
so i just did and i had an issue at first but resolved it with `pnpm install --force`

1. `pnpm install --force` on parent dir
2. `pnpm build`on parent dir
3. `cd playground`and `pnpm dev``

that worked for me on the github codespace
